source/material_model/rheology/elasticity.cc

```diff
@@ -290,6 +291,14 @@ namespace aspect
               std::make_unique<ElasticAdditionalOutputs<dim>> (n_points));
           }
 
+        // We need to modify the shear heating outputs to correctly account for elastic stresses.
```

Comment:
Does this mean the PrescribedShearHeatingOutputs are always created even though shear heating might not be included in the energy equation? The ElasticOutputs are added to the additional_outputs only in the Stokes assembly, maybe we can do a similar thing for the PrescribedShearHeatingOutputs?

Reply:
Ah, I didnt realize that `ElasticOutputs` are only created in the Stokes assembly, while `ElasticAdditionalOutputs` are always created. But then why did the previous code work at all? It would only prescribe the shear heating terms when called from the Stokes assembly where the terms would not be used?

I have looked into modifying the code to only create these outputs in the advection assembly, however it would add a lot of boilerplate code to the different advection assemblers (also for cell and face assemblers), and it would make the heating output in the postprocessor inconsistent with the heating term actually used in the equation. What is your main concern about computing it always? Efficiency? Then I would say we can worry about it in case it appears in any measurements, I dont expect it to be a major cost.